// @storybook/vue/dist/server/framework-preset-vue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.webpack = webpack;
exports.babelDefault = babelDefault;

var _plugin = _interopRequireDefault(require("vue-loader/lib/plugin"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function webpack(config) {
  return Object.assign({}, config, {
    plugins: [...config.plugins, new _plugin.default()],
    module: Object.assign({}, config.module, {
      rules: [...config.module.rules, {
        test: /\.vue$/,
        loader: require.resolve('vue-loader'),
        options: {}
      }]
    }),
    resolve: Object.assign({}, config.resolve, {
      extensions: [...config.resolve.extensions, '.vue'],
      alias: Object.assign({}, config.resolve.alias, {
        vue$: require.resolve('vue/dist/vue.esm.js')
      })
    })
  });
}

function babelDefault(config) {
  return Object.assign({}, config, {
    presets: [...config.presets, require.resolve('babel-preset-vue')]
  });
}
